proto-lens-protoc/src/Data/ProtoLens/Setup.hs

```diff
@@ -66,6 +68,22 @@ defaultMainGeneratingProtos
 defaultMainGeneratingProtos root
     = defaultMainWithHooks $ generatingProtos root simpleUserHooks
 
+-- | This behaves the same as 'Distribution.Simple.defaultMain', but
+-- auto-generates Haskell files from the .proto files listed. The given .proto
+-- files should be under the given root directory.
+--
+-- Writes the generated files to the autogen directory (@dist\/build\/autogen@
+-- for Cabal, and @.stack-work\/dist\/...\/build\/autogen@ for stack).
+--
+-- Throws an exception if the @proto-lens-protoc@ executable is not on the PATH.
+defaultMainGeneratingSpecificProtos
+    :: FilePath -- ^ The root directory under which .proto files can be found.
+    -> [FilePath] -- ^ The .proto files under @root@
```

Comment:
As you mentioned, it may make more sense for these to be relative to the root path, though I'm not sure whether that makes the code below more complicated.  Either way, let's document the behavior when they're not absolute paths.